### PR TITLE
[WIP]run badfish as a container

### LINKED ISSUE #176: Run badfish as container
https://github.com/redhat-performance/badfish#usage-via-docker

### PR DESCRIPTION
**Pull request by asyedham**
fix #176

### CODE REVIEW

File: setup_undercloud.yml [hunk 1]
```diff
@@ -50,19 +49,8 @@
...
-        - name: Create badfish virtualenv
-          command: python3 -m venv {{ badfish_venv }} creates={{ badfish_venv }}
-
-        - name: Install badfish requirements
-          pip:
-            requirements: "{{ ansible_user_dir }}/badfish/requirements.txt"
-            virtualenv_command: /usr/bin/python3 -m venv
-            virtualenv: "{{ badfish_venv }}"
+        - name: pulls docker image of badfish
+          shell: docker pull quads/badfish
```
Comment (smalleni) [inline setup_undercloud.yml:53]: Why not use podman module. Please run using podman.

Comment (smalleni) [reply setup_undercloud.yml:53]: podman_image

File: group_vars/all.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ ansible_ssh_key: "{{ ansible_user_dir }}/.ssh/id_rsa"
 osp_release: 13
 osp_puddle: passed_phase2
 controller_count: 3
+dns_ip: 10.1.32.3
```
Comment (smalleni) [inline group_vars/all.yml:15]: Not needed

Comment (smalleni) [reply group_vars/all.yml:15]: Also you can add badfish_image as a variable and use it when you pull the podman image.

File: setup_undercloud.yml [hunk 3]
```diff
@@ -94,10 +82,7 @@
 
         - name: set undercloud to PXE boot off Foreman (Dell)
           shell: |
-              source {{ badfish_venv }}/bin/activate
-              python3 badfish.py -H mgmt-{{ undercloud_hostname }} -u quads -p {{ chassis_password }} -i config/idrac_interfaces.yml --boot-to-type foreman --pxe
-          args:
-            chdir: "{{ ansible_user_dir }}/badfish"
+              docker run --rm --dns {{ dns_ip }} quads/badfish -H mgmt-{{ undercloud_hostname }} -u quads -p {{ chassis_password }} --pxe
```
Comment (smalleni) [inline setup_undercloud.yml:85]: Use podman and you don't need DNS
For example,
podman run -it --rm {{ badfish_pod_image }} -u  quads-p {{ chassis_assword }} -i config/idrac_interfaces.yml

**Review by smalleni — CHANGES_REQUESTED** [review]
Please make suggested changes.

**Comment by asyedham** [comment]
Closing this PR. As we need to manually install podman packages. (i.e  it may be vary depending on OS)

**Comment by smalleni** [comment]
For all Red Hat based distros it should be yum/dnf . However we can use ansible package module. I don't think we need to deal with repos (it's beyond the scope)

**Comment by asyedham** [comment]
For Red Hat based distros we need to enable extra repo  to install podman e.g for RHEL 7 ``rhel-7-server-extras-beta-rpms``

**Pull request closed without merging**